impl/src/main/java/org/jboss/forge/roaster/model/impl/MethodImpl.java

```diff
@@ -719,12 +719,8 @@ else if (!getOrigin().hasImport(type) && !getOrigin().hasImport(thrown))
    @Override
    public ParameterSource<O> addParameter(String type, String name)
    {
-      if (getOrigin().requiresImport(type))
-      {
-         getOrigin().addImport(type);
-      }
-      String stub = "public class Stub { public void method( " + Types.toSimpleName(Types.fixArray(type, false)) + " "
-               + name + " ) {} }";
+      String resolvedType = TypeImpl.ensureImports(getOrigin(), new TypeImpl<O>(getOrigin(),null,type));
+      String stub = "public class Stub { public void method( " + resolvedType + name + " ) {} }";
```

Comment:
Tests fail just because a + " " + was lost between resolvedType and name :)
If we agree on the rest, I'll amend the PR